dotnet/src/SemanticKernel.UnitTests/SkillDefinition/SKFunctionTests2.cs

```diff
@@ -155,238 +150,268 @@ static string Test(SKContext cx)
 
         // Assert
         Assert.False(result.ErrorOccurred);
-        this.VerifyFunctionTypeMatch(5);
-        Assert.Equal(s_expected, s_canary);
+        Assert.Equal(s_expected, s_actual);
         Assert.Equal("abc", context.Result);
     }
 
     [Fact]
-    public async Task ItSupportsType5NullableAsync()
+    public async Task ItSupportsInstanceContextStringNullableAsync()
     {
         // Arrange
+        int invocationCount = 0;
+
         [SKFunction("Test")]
         [SKFunctionName("Test")]
         string? Test(SKContext cx)
         {
-            s_canary = cx["someVar"];
+            invocationCount++;
+            s_actual = cx["someVar"];
             return "abc";
         }
 
         var context = this.MockContext("");
         context["someVar"] = s_expected;
 
         // Act
-        var function = SKFunction.FromNativeMethod(Method(Test), log: this._log.Object);
+        Func<SKContext, string?> method = Test;
+        var function = SKFunction.FromNativeMethod(Method(method), method.Target, log: this._log.Object);
         Assert.NotNull(function);
         SKContext result = await function.InvokeAsync(context);
 
         // Assert
         Assert.False(result.ErrorOccurred);
-        this.VerifyFunctionTypeMatch(5);
-        Assert.Equal(s_expected, s_canary);
+        Assert.Equal(1, invocationCount);
```

Comment:
how come we need to check the count in this and other tests, but not all tests?

Reply:
It'd be nice to check in them all, but it's much more complicated to do so in one's that are static and can't close over instance state. It'll be easy to improve once variables can be arbitrary objects.